### PR TITLE
Update to predefined meshes (Capytaine v2)

### PR DESCRIPTION
**Pull request by jtgrasb**
## Description
This PR updates WecOptTool tests to use predefined meshes from Capytaine instead of the predefined bodies (which [are deprecated](https://github.com/capytaine/capytaine/blob/master/docs/changelog.rst#new-in-version-15-2022-12-13))

## Type of PR
- [ ] Bug fix
- [X] New feature
- [ ] Documentation
- [ ] Other: (specify)

## Checklist for PR
- [X] Authors read the [contribution guidelines](https://github.com/sandialabs/WecOptTool/blob/main/.github/CONTRIBUTING.md)
- [X] The pull request is from an issue branch (not main) on *your* fork, to the [main branch in WecOptTool](https://github.com/sandialabs/WecOptTool).
- [X] The authors have given the admins edit access
- [X] All changes adhere to the style guide including PEP8, Docstrings, and Type Hints.
- [ ] Modified the documentation if applicable
- [X] Modified or added a new test
- [ ] All tests pass
- [ ] [Reference or close any relevant issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)

### CODE REVIEW

File: tests/test_hydrostatics.py [hunk 1]
```diff
@@ -55,8 +55,10 @@ def mass(lx, ly, lz, rho):
 @pytest.fixture()
 def fb(lx, ly, lz):
     """Simple constant density rectangular barge."""
-    rect = cpy.RectangularParallelepiped(
-        (lx, ly, lz), resolution=(100, 100, 10), center=(0.0, 0.0, 0.0,))
+    rect_mesh = cpy.mesh_parallelepiped(
+            size=(lx, ly, lz), resolution=(100, 100, 10), center=(0.0, 0.0, 0.0,)
+        )
```
Comment (cmichelenstrofer) [inline tests/test_hydrostatics.py:60]: Try to be consistent with the tabs/alignment in multi-line function calls. I say look at whatever is most common the file/rest of code.

Comment (michaelcdevin) [reply tests/test_hydrostatics.py:60]: Honestly the code is pretty inconsistent on this... I think as long as the `test_hydrostatics` and `test_core` versions of this line are the same and don't exceed the max line lengths it should be good :)

**Review by michaelcdevin — APPROVED** [review]
This looks good! Go ahead and merge once the multiline styling comment from @cmichelenstrofer is resolved.

File: tests/test_hydrostatics.py (hunk 1 above)
Comment (cmichelenstrofer) [reply tests/test_hydrostatics.py:60]: sounds good

Comment (jtgrasb) [reply tests/test_hydrostatics.py]: updated to be more consistent based on other multi-line calls in `test_core`

**Pull request merged**